src/models/v2/message-trait.ts

```diff
@@ -0,0 +1,98 @@
+import { BaseModel } from "../base";
+import { CorrelationId } from './correlation-id';
+import { MessageExamples } from './message-examples';
+import { MessageExample } from './message-example';
+import { Schema } from './schema';
+
+import { Mixin } from '../utils';
+import { BindingsMixin } from './mixins/bindings';
+import { DescriptionMixin } from './mixins/description';
+import { ExtensionsMixin } from './mixins/extensions';
+import { ExternalDocumentationMixin } from './mixins/external-docs';
+import { TagsMixin } from './mixins/tags';
+
+import type { ModelMetadata } from "../base";
+import type { CorrelationIdInterface } from "../correlation-id";
+import type { MessageExamplesInterface } from "../message-examples";
+import type { MessageTraitInterface } from "../message-trait";
+import type { SchemaInterface } from "../schema";
+
+export class MessageTrait extends Mixin(BaseModel, BindingsMixin, DescriptionMixin, ExtensionsMixin, ExternalDocumentationMixin, TagsMixin) implements MessageTraitInterface {
+  constructor(
+    private readonly _id: string,
+    _json: Record<string,any>,
+    _meta: ModelMetadata = {} as any
+  ) {
+    super(_json, _meta);
+  }
+
+  id(): string {
+    return this.messageId() || this._id;
```

Comment:
Noice!!! 👍

Reply:
Here I have a problem, because on one side we call it `id`, and next to it we have `messageId`. Maybe we should rename this method to `uid` and for `id` return the value for `messageId`? Same for other objects - where we call the method `id` we should call it `uid`?

Reply:
The `messageId` is the real **unique id** (uid) across the document, so I think it would be the other way around.
Anyway, I see the issue you are mentioning. 
Maybe, adding a new `UniqueIdentifiableInterface`, with the `uid` method would make sense. In the case of message, to return `this.messageId()`, `this.operationId` for Operation, etc.
The `id` field will then just return the key of the map which the object belongs to. 

Makes sense?

Reply:
It makes sense, but I wonder if we need 3 methods, `messageId`, `id` and `uid`. 

The `id` could be for `messageId`
The `uid` for the `messageId` or key or some generated unique id for given reference.

Reply:
I think we are considering a different thing for what the `id` value is.

From my pov, `id` is just the key on the map. For example, `myMessage` in the following example:

```yaml
components:
  messages:
    myMessage:
      # etc
```

Of course, currently messages declared inside the operation won't have an id (at least on spec v2.x) but this could change for next versions (and makes sense to me).